pipelines/foreman_split_puppet.yml

```diff
@@ -0,0 +1,109 @@
+- hosts: localhost
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - forklift
+
+- hosts:
+  - "{{ forklift_server_name }}"
+  - "{{ forklift_proxy_name }}"
+  - "{{ forklift_puppet_name }}"
+  become: yes
+  vars:
+    foreman_installer_skip_installer: true
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - role: forklift_versions
+      scenario: "{{ pipeline_type }}"
+      scenario_version: "{{ pipeline_version }}"
+    - role: foreman_server_repositories
+    - role: update_os_packages
+    - role: foreman_installer
+
+- hosts: "{{ forklift_puppet_name }}"
+  become: yes
+  vars_files:
+    - vars/puppet_base.yml
+  roles:
+    - role: forklift_versions
+      scenario: "{{ pipeline_type }}"
+      scenario_version: "{{ pipeline_version }}"
+    - role: foreman_installer
+      foreman_installer_options_internal_use_only:
+        - "--no-enable-foreman"
+        - "--no-enable-foreman-cli"
+        - "--no-enable-foreman-proxy"
+        - "--puppet-server true"
+        - "--puppet-server-foreman-url https://{{ forklift_server_name }}.{{ ansible_domain }}"
+        - "--puppet-server-admin-api-whitelist localhost"
+        - "--puppet-server-admin-api-whitelist {{ forklift_puppet_name }}.{{ ansible_domain }}"
+        - "--puppet-server-admin-api-whitelist {{ forklift_proxy_name }}.{{ ansible_domain }}"
+        - "--puppet-server-ca-client-whitelist localhost"
+        - "--puppet-server-ca-client-whitelist {{ forklift_puppet_name }}.{{ ansible_domain }}"
+        - "--puppet-server-ca-client-whitelist {{ forklift_proxy_name }}.{{ ansible_domain }}"
+        - "--puppet-autosign-entries {{ forklift_server_name }}.{{ ansible_domain }}"
+        - "--puppet-autosign-entries {{ forklift_proxy_name }}.{{ ansible_domain }}"
+
+# TODO: set foreman's trusted hosts to [{{ forklift_puppet_name }}.{{ ansible_domain }}]
```

Comment:
What prevents this TODO?

Reply:
Haven't looked at automating it. It could probably be automated by calling `foreman-rake config -- -k trusted_hosts -v ...` but I was also thinking about a cleaner solution so I left it in for now.